projects/packages/account-protection/src/class-password-detection.php

```diff
@@ -0,0 +1,386 @@
+<?php
+/**
+ * Class used to define Password Detection.
+ *
+ * @package automattic/jetpack-account-protection
+ */
+
+namespace Automattic\Jetpack\Account_Protection;
+
+use Automattic\Jetpack\Connection\Client;
+use Automattic\Jetpack\Connection\Manager as Connection_Manager;
+
+/**
+ * Class Password_Detection
+ */
+class Password_Detection {
+	const PASSWORD_DETECTION_USER_META_KEY = 'jetpack_account_protection_password_status';
+
+	/**
+	 * Password reset email dependency.
+	 *
+	 * @var Password_Reset_Email
+	 */
+	private $password_reset_email;
+
+	/**
+	 * Password_Detection constructor.
+	 *
+	 * @param ?Password_Reset_Email $password_reset_email Password reset email instance.
+	 */
+	public function __construct( ?Password_Reset_Email $password_reset_email = null ) {
+		$this->password_reset_email = $password_reset_email ?? new Password_Reset_Email();
+	}
+
+	/**
+	 * Redirect to the password detection page.
+	 *
+	 * @return string The URL to redirect to.
+	 */
+	public function password_detection_redirect(): string {
+		return home_url( '/wp-login.php?action=password-detection' );
+	}
+
+	/**
+	 * Check if the password is safe after login.
+	 *
+	 * @param \WP_User|\WP_Error $user The user or error object.
+	 * @param string             $password The password.
+	 * @return \WP_User|\WP_Error The user object.
+	 */
+	public function login_form_password_detection( $user, string $password ) {
+		// Check if the user is already a WP_Error object
+		if ( is_wp_error( $user ) ) {
+			return $user;
+		}
+
+		// Ensure the password is correct for this user
+		if ( ! wp_check_password( $password, $user->user_pass, $user->ID ) ) {
+			return $user;
+		}
+
+		if ( ! $this->validate_password( $password ) ) {
+			// TODO: Ensure usermeta is always up to date
+			$this->update_usermeta( $user->ID, 'unsafe' );
+
+			// Redirect to the password detection page
+			add_filter( 'login_redirect', array( $this, 'password_detection_redirect' ), 10, 3 );
+		} else {
+			$this->update_usermeta( $user->ID, 'safe' );
+		}
+
+		return $user;
+	}
+
+	/**
+	 * Render password detection page.
+	 */
+	public function render_page() {
```

Comment:
Using `void` or `never` as `@return` values and in type hinting both result in separate static analysis error. Interested in a second opinion as to how to surpass this issue.

Reply:
Disregard, the correct combination was using `never` in the docs with no return type hinting.